packages/ui/src/components/va-tabs/VaTab/VaTab.vue

```diff
@@ -39,111 +40,152 @@
-
-import ColorMixin from '../../../services/color-config/ColorMixin'
-import useKeyboardOnlyFocus from '../../../composables/useKeyboardOnlyFocus'
-import { RouterLinkMixin } from '../../../mixins/RouterLinkMixin/RouterLinkMixin'
+import {
+  computed,
+  defineComponent,
+  inject,
+  onBeforeUnmount,
+  onMounted,
+  ref,
+  unref,
+} from 'vue'
 import VaIcon from '../../va-icon'
+import { useRouterLink, useRouterLinkProps } from '../../../composables/useRouterLink'
+import useKeyboardOnlyFocus from '../../../composables/useKeyboardOnlyFocus'
+import { useColor } from '../../../composables/useColor'
+import { TabsViewKey, TabsView, TabComponent } from '../types'
 
-import { TabsServiceKey, TabsService } from '../VaTabs.vue'
-
-type Context = {
-  tabsService: TabsService;
-  hasKeyboardFocus: boolean;
-  keyboardFocusListeners: Record<string, () => void>;
-}
-
-class TabProps {
-  color = prop<string>({ type: String, default: undefined })
-  icon = prop<string>({ type: String, default: null })
-  label = prop<string>({ type: String, default: null })
-  disabled = prop<boolean>({ type: Boolean })
-  name = prop<string | number>({ type: [String, Number] })
-  tag = prop<string>({ type: String, default: 'div' })
-}
-
-const TabPropsMixin = Vue.with(TabProps)
-
-@Options({
+export default defineComponent({
   name: 'VaTab',
   components: { VaIcon },
   emits: ['click', 'keydown-enter', 'focus'],
-})
-export default class VaTab extends mixins(
-  RouterLinkMixin,
-  TabPropsMixin,
-  ColorMixin,
-) {
-  id: string | number | null = null
-  isActive = false
-  rightSidePosition = 0
-  leftSidePosition = 0
-  hoverState = false
-
-  context: Context = setup(() => {
-    const tabsService = inject(TabsServiceKey) as TabsService
 
+  props: {
+    ...useRouterLinkProps,
+    color: {
+      type: String,
+      default: '',
+    },
+    icon: {
+      type: String,
+      default: '',
+    },
+    label: {
+      type: String,
+      default: '',
+    },
+    disabled: {
+      type: Boolean,
+    },
+    name: {
+      type: [String, Number],
+    },
+    tag: {
+      type: String,
+      default: 'div',
+    },
+  },
+
+  setup: (props, { emit }) => {
+    const tabElement = ref<HTMLElement | null>(null)
+
+    const isActive = ref(false)
+    const hoverState = ref(false)
+    const rightSidePosition = ref(0)
+    const leftSidePosition = ref(0)
     const { hasKeyboardFocus, keyboardFocusListeners } = useKeyboardOnlyFocus()
-
-    return {
-      tabsService,
-      hasKeyboardFocus,
-      keyboardFocusListeners,
+    const { tagComputed, hrefComputed, isActiveRouterLink } = useRouterLink(props)
+    const { colorComputed } = useColor(props)
+    const classComputed = computed(() => ({ 'va-tab--disabled': props.disabled }))
+    const {
+      parentDisabled,
+      selectTab,
+      moveToTab,
+      registerTab,
+      unregisterTab,
+    } = inject(TabsViewKey) as TabsView
```

Comment:
```suggestion
    } = inject<TabsView>(TabsViewKey)
```

https://vuejs.org/guide/typescript/composition-api.html#typing-provide-inject

Reply:
I don't want to provide a default value by declaring the generic type like in your example.